src/pages/MyProfile.jsx

```diff
@@ -1,15 +1,240 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
+import PopupJS from "reactjs-popup";
 import PageLayout from "../components/PageLayout";
 import Header from "../components/Header";
 import Footer from "../components/Footer";
+import api from "../http/api";
+
+import 'reactjs-popup/dist/index.css';
+import hide from "../assets/hide-pass.svg";
+import show from "../assets/show-pass.svg";
+import {useNavigate, useParams} from "react-router-dom";
+import {toast} from "react-toastify";
 
 function MyProfile() {
+
+    const [profileData, setProfileData] = useState({
+        id: '',
+        username: '',
+        firstName: '',
+        lastName: '',
+        roles: ''
+    });
+
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                await api.get('/stats/user', {
+                    headers: {
+                        'Authorization': 'Bearer ' + localStorage.getItem('token')
+                    }
+                }).then( (res) => setProfileData(res.data));
+               } catch (error) {}
+        };
+
+        fetchData();

```

Comment:
remove catch and try from upper function fetchData and set .catch when you call method in this 36. line.